Sami/Parser/ClassVisitor/PropertyClassVisitor.php

```diff
@@ -50,58 +49,26 @@ public function visit(ClassReflection $class)
      * Adds a new property to the class using an array of tokens.
      *
      * @param ClassReflection $class       Class reflection
-     * @param string          $propertyTag Property tag contents
+     * @param string|array    $propertyTag Property tag contents
      *
      * @return bool
      */
     protected function injectProperty(ClassReflection $class, $propertyTag)
     {
-        if (!$data = $this->parseProperty($propertyTag)) {
-            return false;
-        }
-
-        $property = new PropertyReflection($data['name'], $class->getLine());
-        $property->setDocComment($data['description']);
-        $property->setShortDesc($data['description']);
-
-        if (isset($data['hint'])) {
-            $property->setHint(array(array($data['hint'], null)));
-        }
-
-        $class->addProperty($property);
-
-        return true;
-    }
-
-    /**
-     * Parses the parts of an @property tag into an associative array.
-     *
-     * @param string $tag Property tag contents
-     *
-     * @return array
-     */
-    protected function parseProperty($tag)
-    {
-        // Account for default array syntax
-        $tag = str_replace('array()', 'array', $tag);
-
-        /** @var PropertyTag $propertyTag */
-        $propertyTag = $this->context->getDocBlockParser()->getTag('@property '.$tag);
-        $propertyName = $propertyTag->getVariableName();
+        if (is_array($propertyTag) && count($propertyTag) == 3 && !empty($propertyTag[1])) {
```

Comment:
1. You're right of course, should've done that.
2. I may be wrong here, correct me if I am. But: because of the changes to `DocBlockParser.php`, `@property` Tags do NOT consist of a single string anymore. They always get parsed as an array consisting of 3 elements, of which the first is the typehint (null if absent), the second is the variable name and the third is the description (null if absent). So there _should_ be no chance of anything getting lost here, I also testet that on various projects. But again, I might be mislead here, correct me if that is the case.